### PR TITLE
Make ConnectionStatusChangeCallback a settable property

### PR DESCRIPTION
**Pull request by schoims**

### CODE REVIEW

File: iothub/device/src/IotHubBaseClient.cs [hunk 1]
```diff
@@ -106,17 +103,18 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
...
+        /// <example>
+        /// deviceClient.SetConnectionStatusChangeCallback = OnConnectionStatusChanged;
+        /// //...
+        ///
+        /// public void OnConnectionStatusChanged(ConnectionStatusInfo connectionStatusInfo)
+        /// {
+        ///     // Add connection status changed logic as needed
+        /// }
+        /// </example>
+        public Action<ConnectionStatusInfo> SetConnectionStatusChangeCallback { get; set; }
```
Comment (schoims) [inline iothub/device/src/IotHubBaseClient.cs:117]: remove Set

File: iothub/device/src/IotHubBaseClient.cs [hunk 2]
```diff
@@ -106,17 +103,18 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
...
-        /// <param name="statusChangeHandler">The callback for the connection status change notifications.</param>
-        public void SetConnectionStatusChangeCallback(Action<ConnectionStatusInfo> statusChangeHandler)
-        {
-            if (Logging.IsEnabled)
-                Logging.Info(this, statusChangeHandler, nameof(SetConnectionStatusChangeCallback));
-
-            _connectionStatusChangeCallback = statusChangeHandler;
-        }
+        /// <example>
+        /// deviceClient.SetConnectionStatusChangeCallback = OnConnectionStatusChanged;
```
Comment (schoims) [inline iothub/device/src/IotHubBaseClient.cs:109]: remove set

**Comment by schoims** [comment]
\azp run

File: iothub/device/src/IotHubBaseClient.cs [hunk 3]
```diff
@@ -106,17 +103,18 @@ public void SetRetryPolicy(IRetryPolicy retryPolicy)
...
+        /// <example>
+        /// deviceClient.ConnectionStatusChangeCallback = OnConnectionStatusChanged;
+        /// //...
+        ///
+        /// public void OnConnectionStatusChanged(ConnectionStatusInfo connectionStatusInfo)
+        /// {
+        ///     // Add connection status changed logic as needed
+        /// }
+        /// </example>
+        public Action<ConnectionStatusInfo> ConnectionStatusChangeCallback { get; set; }
```
Comment (drwill-ms) [inline iothub/device/src/IotHubBaseClient.cs:117]: Class member order:
- fields
- ctors
- properties
- methods

**Review by drwill-ms — APPROVED** [review]

**Review by abhipsaMisra — APPROVED** [review]

**Comment by schoims** [comment]
\azp run

**Pull request merged**